### PR TITLE
Updating Pyomo version to the 6.2 release

### PR DESCRIPTION
**Pull request by blnicho**
This updates the Pyomo tag to the 6.2 release. I also updated the supported versions of Python in `setup.py` and the `README` because I noticed they were out of date. Looks like we are running tests for Python 3.9.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by jsiirola — DISMISSED** [review]

**Review by lbianchi-lbl — APPROVED** [review]
Looks good, and thanks for updating the README as well!

**Review by mrmundt — APPROVED** [review]

**Review by jsiirola — APPROVED** [review]

**Review by eslickj — APPROVED** [review]

File: idaes/generic_models/properties/core/examples/tests/test_ASU_PR.py [hunk 1]
```diff
@@ -44,6 +44,11 @@
 # Get default solver for testing
 solver = get_solver()
 
+def _as_quantity(x):
```
Comment (andrewlee94) [inline idaes/generic_models/properties/core/examples/tests/test_ASU_PR.py:47]: Seeing as this is turning up in a few places, should we be making this a general utility somewhere?

**Review by andrewlee94 — APPROVED** [review]
Only one minor question that doesn't need to hold this up.

File: idaes/generic_models/properties/core/examples/tests/test_ASU_PR.py (hunk 1 above)
Comment (jsiirola) [reply idaes/generic_models/properties/core/examples/tests/test_ASU_PR.py:47]: Yes ... probably in Pyomo.  I am working on that walker, but it wasn't close enough to make yesterday's release deadline.  I figured the easiest thing was to put the temporary (inefficient) method here and then replace it once the efficient one was implemented.

**Comment by eslickj** [comment]
@lbianchi-lbl, do you know why the integration tests don't seem to be running on this PR?

File: idaes/generic_models/properties/core/examples/tests/test_ASU_PR.py (hunk 1 above)
Comment (andrewlee94) [reply idaes/generic_models/properties/core/examples/tests/test_ASU_PR.py:47]: I was guessing that was the case, but wanted to confirm.

**Pull request merged**